### ISSUE TITLE
matmodlab2 installation 

### ISSUE THREAD

**Issue by Qalkaseasbeh** [matmodlab/matmodlab2#11]
Hello, 

I'm still new to this tool, so my question is very simple.

I downloaded matmodlab, and (prerequisites: python, numpy, and scipy), but I don't know how to install matmodlab itself, I cannot understand how it works. could you help please?

Thanks a lot.

**Comment by sswan** [comment]
It should be enough to download the zip file, unzip it, then set your working directory to the extracted folder. There is no need to install, so long as your working directory includes the matmodlab2 directory.

**Comment by Qalkaseasbeh** [comment]
You mean Python directory (working directory), right?

**Comment by sswan** [comment]
Yes, the PYTHONPATH.

**Comment by Qalkaseasbeh** [comment]
Would you mind telling how I can do that?

**Comment by sswan** [comment]
Without knowing your specific setup, your best best would be to follow the instructions at this website:

http://stackoverflow.com/questions/3701646/how-to-add-to-the-pythonpath-in-windows-7

However, if you are using an IDE (such as Canopy) there is an option to set the working directory (which is slightly different from the PYTHONPATH, but would do the same thing for you).

**Comment by Qalkaseasbeh** [comment]
Thanks a lot for your help. Let me try with this.

**Comment by tjfulle** [comment]
Thanks @Qalkaseasbeh for the questions/feedbacks!

At its heart, `matmodlab2` is just a python package.  So, "installation" can be as easy as cloning the repository to wherever you like and adding the directory to the `PYTHONPATH` environment variable.  For example, suppose you clone (or download and unzip) `matmodlab2` to `/home/user/Documents/matmodlab2`, then in your bash shell just do:

```
export PYTHONPATH=/home/user/Documents/matmodlab2
```

or in `csh`

```
setenv PYTHONPATH /home/user/Documents/matmodlab2
```

On windows, you have to go to 

```
Computer > Properties > Advanced system settings > Environment Variables
``` 

and then edit or add the path to `matmodlab2` to the `PYTHONPATH`.  See [How do I set or change the PATH system variable?](https://www.java.com/en/download/help/path.xml) for more details.

I prefer to work in a terminal environment, but I believe that most python IDE's (like Spyder) allow you to set the working directory or `PYTHONPATH` directly in their settings.

To get a start with `matmodlab2`, I *highly* recommend (after setting the `PYTHONPATH`) looking at the notebooks in `matmodlab2/notebooks`.  To look at the notebooks, you need to have Jupyter notebook installed.  The notebooks are live tutorials of how to use `matmodlab2` and there are many examples of material models.  There is also a notebook on writing Abaqus UMATs.  If you don't have Jupyter installed, I highly recommend installing [Anaconda](www.continuum.io) and using their python distribution.  The Anaconda Launcher program can launch Jupyter notebooks and Spyder IDE sessions.  If you don't have Jupyter, the notebooks can be viewed directly on github.

**Comment by tjfulle** [comment]
@Qalkaseasbeh, I would also recommend cloning and not downloading - I push through changes often and by cloning you can easily pull down the changes.

**Comment by Qalkaseasbeh** [comment]
Thanks a lot for your deep explanation. 

I did add the path to matmodlab2 to the PYTHONPATH, but How can I check it's done perfectly?
In the second comment, you said " I would also recommend cloning and not downloading", and I wanna keep updated with changes you make, but how can I do that?

I did download ANACONDA. I'm totally new for all these stuff and I'll follow your recommendations to start.

Once again, Thank you.

**Comment by tjfulle** [comment]
@Qalkaseasbeh said

> I did add the path to matmodlab2 to the PYTHONPATH, but How can I check it's done perfectly?

> I did download ANACONDA. I'm totally new for all these stuff and I'll follow your recommendations to start.

So, open the Anaconda Launcher application, launch the QT Console app and execute

>>> import sys
>>> print(sys.path)

If you don't see the Matmodlab2 path in the resultant list, you did not do it right.


> In the second comment, you said " I would also recommend cloning and not downloading", and I wanna keep updated with changes you make, but how can I do that?

Google "how to set up git on windows".  After getting it set up, launch the git bash console and

```
cd <path where you want matmodlab>
git clone https://www.github.com/matmodlab/matmodlab2 
```

Make sure *this* path is in your `PYTHONPATH`.

I'd then go through the Matmodlab2 tutorials with the Jupyter notebook app that is bundled with Anaconda.

**Comment by Qalkaseasbeh** [comment]
I did what you recommended:

1- So, open the Anaconda Launcher application, launch the QT Console app and execute

import sys
print(sys.path)

Me: I see the path where I downloaded matmodlab2 ('C:\\Users\\qusay.alkaseasbeh\\Documents') as attached in picture. 

![capture](https://cloud.githubusercontent.com/assets/24799302/24085728/fc6b4976-0cce-11e7-9ccc-b850dccf086d.PNG)

2- Google "how to set up git on windows". After getting it set up, launch the git bash console and

cd <path where you want matmodlab>
git clone https://www.github.com/matmodlab/matmodlab2 
Make sure this path is in your PYTHONPATH.

Me: I did, but It didn't accept the path. see the attached please. 
![git](https://cloud.githubusercontent.com/assets/24799302/24085768/af8577f2-0ccf-11e7-90ab-afb6fd628849.PNG)

**Comment by tjfulle** [comment]
The error is pretty clear - you already have a directory named matmodlab2 in that directory.  You'll have to change its name or remove it before cloning

**Comment by Qalkaseasbeh** [comment]
Yeah, there is unzip file in this folder ('C:\Users\qusay.alkaseasbeh\Documents'), but it doesn't accept any path doesn't have this unzip file. 

Queation: since I got done with first method, could I start studying the examples?  

Sorry for disturbing

**Comment by tjfulle** [comment]
Yes, you can use that version.  I have made updates to the notebooks in the last two days, so you may want to download a newer version

**Comment by Qalkaseasbeh** [comment]
I think everything looks good now, right? Taylor (Technical support at UND) helped me to get this stage. 

Thanks a lot tjfulle  for the time you spent to help me yesterday.

![capture](https://cloud.githubusercontent.com/assets/24799302/24109681/6d0904d0-0d5f-11e7-98c9-47739b844e0b.PNG)
![git](https://cloud.githubusercontent.com/assets/24799302/24109682/6d0e2e06-0d5f-11e7-9f86-39f6a7ac8e6c.PNG)

**Comment by Qalkaseasbeh** [comment]
Hello,

I've started practicing some examples but I'm running into this issue (attached). Could you help please?
![capture](https://cloud.githubusercontent.com/assets/24799302/24299730/20fbb04e-1078-11e7-97a7-a3eee82d4ae2.PNG)

**Comment by tjfulle** [comment]
Your `PYTHONPATH` is probably incorrect.  See our previous conversations.

**Comment by Qalkaseasbeh** [comment]
I checked it with (import sys, print(sys.path) and it's supposed to be okay (I attached what I got 4 days ago).

**Comment by tjfulle** [comment]
`matmodlab2` is just a python package, so if your `PYTHONPATH` is correct, it will load.  You can try launching the terminal session from the `matmodlab2` directory.  Or, see if you can set the `PYTHONPATH` from the QT console's preferences.  I've never used the Jupyter console, but that's what I would do to start.

**Comment by Qalkaseasbeh** [comment]
" You can try launching the terminal session from the matmodlab2 directory" How can I do it?

**Comment by tjfulle** [comment]
I believe you make a shortcut to the app and then change the properties of the shortcut to launch where you want.  I'm not a windows user, that's about all I know.

The best thing to do is figure out why `Matmodlab2` is not being imported.
